### PR TITLE
Add optional watermarks

### PR DESCRIPTION
**Pull request by micnem**
This PR introduces an optional watermark feature to the video generation process. Users can now include a watermark image URL in the input script, and the watermark will be added to the final video output.

Key changes include:

- Update to the README.md to include an example of specifying a watermark URL in the input script.

- Addition of a new watermark.py file containing the add_watermark function, which adds the watermark to the video.

- Modification of the video_generator.py file to integrate the watermark feature if a watermark URL is provided in the input script.

The new watermark feature allows users to customize their video outputs with their desired watermark images, providing a professional touch and protecting their content.

### CODE REVIEW

**Review by fitzhavey — APPROVED** [review]

File: scripts/watermark.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from moviepy.editor import *
+import requests
+import os
+import requests
```
Comment (dewinterjack) [inline scripts/watermark.py]: line 4 and 5 should be deleted

File: scripts/watermark.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    # Write the video clip to a file
+    video_with_watermark.write_videofile(output_video_path)
+
+    # Close the clips to free up memory
+    clip.close()
+    video_with_watermark.close()
+
+
+def download_img(url):
```
Comment (dewinterjack) [inline scripts/watermark.py:30]: why are we downloading a watermark image when we can just provide a local path?

**Review by dewinterjack — CHANGES_REQUESTED** [review]

File: scripts/watermark.py (hunk 2 above)
Comment (fitzhavey) [reply scripts/watermark.py:30]: My thoughts are a given video might have any given watermark. If we use this engine to create content for multiple channels they might need multiple separate watermarks.

File: README.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ The project uses the Eleven Labs API to generate the voice narration and MoviePy
 Here is a simple example script that can be passed as an input to the app:
 ```yaml
 title: Simple Video Example
+watermark: "https://i.imgur.com/NbzMg2q.png" ## Optional
```
Comment (fitzhavey) [inline README.md]: Double ## before comment, only need the 1

**Review by fitzhavey — CHANGES_REQUESTED** [review]

**Review by fitzhavey — APPROVED** [review]

**Review by dewinterjack — APPROVED** [review]

**Pull request merged**